lib/src/presentation/bloc/card_bloc.dart

```diff
@@ -0,0 +1,100 @@
+import 'dart:async';
+
+import '../../core/util/card_constants.dart';
+import '../../data/model/card.dart';
+import '../../domain/repository/repository_card.dart';
+import 'interfaces/i_card_bloc.dart';
+
+class CardBloc extends ICardBloc {
+  final RepositoryCard repositoryCard;
+  final defaultEndpoint = kRacesDemonEndPointApi;
+  late int firstCardOfPage = cardBlocStartingPositionOfShowedCards;
+  late int lastCardOfPage = kCardBlocNumberOfCardsShowedPerPage;
+  int currentPage = kCardBlocStartingCurrentPage;
+  late List<HearthstoneCard> allCards = [];
+  int numberOfPagesCardList = kDefaultNumberOfPagesCardList;
+  bool isLoading = false;
+
+  CardBloc({required this.repositoryCard});
+
+  final StreamController<List<HearthstoneCard>> _streamController =
+      StreamController<List<HearthstoneCard>>.broadcast();
+
+  @override
+  void dispose() {
+    _streamController.close();
+  }
+
+  @override
+  Future<void> initialize() async {
+    await sinkCardList(
+      defaultEndpoint,
+    );
+  }
+
+  @override
+  Stream<List<HearthstoneCard>> get cardListStream => _streamController.stream;
+
+  @override
+  Future<void> sinkCardList(String endpoint) async {
+    isLoading = true;
+    currentPage = kCardBlocStartingCurrentPage;
```

Comment:
Where do you use this two variables?

Reply:
I use isLoading in the function sinkCardList(endpoint)   (card_bloc line 39)  and in the card_list in the StreamBuilder<>, to tell the UI to display a CircularProgressIndicator when the function  fillAllCardsList()   (card_bloc line 42) is still in progress. 
I use currentPage in the function sinkCurrentCardList(),  sinkNextCardList() and sinkPreviousCardList()  (card_bloc line 57, line 78, line 85 )  To calculate the firstCardOfPage  and to check that it doesn't go off range for the buttons next and previous page.  Both of the ElevattedButtons in (card_list) and also to update the currentPage number in screen.  (card_list line 38, etc)

Reply:
isLoading is really used on the UI?

Reply:
<img width="450" alt="image" src="https://user-images.githubusercontent.com/109122811/188971641-4978758c-aa96-498a-8f93-6af1e71d9658.png">
It's used this way...

Reply:
But you really used it or you use ConnectionState.waiting?